_posts/pl/2021-01-28-migracja-schematow-bazy-danych.md

```diff
@@ -0,0 +1,108 @@
+---
+layout:    post
+title:     Migracja schematów bazy danych
+date:      2021-01-28 08:00:00 +0100
+published: false
+lang:      pl
+author:    jgoszczurny
+image:     /assets/img/posts/2021-01-28-migracja-schematow-bazy-danych/bird-migrations.jpg
+tags:
+    - migration
+    - database
+    - liquibase
+    - flyway
+    - kubernetes
+    - job
+    - initContainers
+---
+
+Niemożliwe jest rozwijanie aplikacji bez równoczesnego rozwijania schematu bazy danych.
+
+Tak jak podczas rozwijania naszej aplikacji, stosujemy wzorce projektowe przy pisaniu kodu, tak w przypadku rozwijania schematów baz danych również powinniśmy stosować się do takich wytycznych, aby rozwijanie bazy danych było przyjemnością, a nie ostatecznością
+
+## Migracja schematu, czyli ewolucyjny projekt bazy danych (eng. Evolutionary Database Design)
+Aby umożliwić proste rozwijanie naszej bazy danych, możemy skorzystać z przetestowanego już zbioru zaleceń [🔗¹](https://www.martinfowler.com/articles/evodb.html) [🔗²](https://en.wikipedia.org/wiki/Evolutionary_database_design), dzięki którym zmiany będą mniej inwazyjne, a nawet bezprzerwowe (zero downtime deployment).
+
+### Przechowywanie zmian w repozytorium kodów
+Pod tym pojęciem kryje się nie tylko trzymanie zmian w repozytorium, ale to właśnie w repozytorium powinno się trzymać wszystkie zmiany dotyczące tej bazy w jednym miejscu.
+
+Jeśli kilka projektów korzysta z bazy, to mimo wszystko zmiany powinny odbywać się tylko na jednym głównym repozytorium (nawet na osobnym).
+
+Dzięki temu mamy możliwość zweryfikowania zmian i odtworzenia bazy (np. lokalnie), dodatkowo zmiany trzymane w jednym miejscu umożliwiają prostsze utrzymywanie kolejności. Dzięki temu uzyskamy zawsze dokładnie ten sam schemat bazy.
+
+### Każda zmiana schematu powinna być migracją
+Każdorazowe modyfikowanie schematu, powinno odbywać się za pomocą migracji, którą zapiszemy w repozytorium, unikając ręcznego modyfikowania schematu.
+
+Dzięki temu nie będzie sytuacji, w której po odtworzeniu bazy, będzie się ona różnić od oryginału.
+
+Dodatkowo zmiany wykonane na boku, mogą wpłynąć na późniejsze jego wykonanie za pomocą migracji, np. wykonujemy CREATE TABLE bezpośrednio na bazie, a później dodajemy migrację schematu, która to robi, w takim wypadku otrzymamy błąd, że taka tabela już istnieje.
```

Comment:
Chciałbym zaproponować jeszcze jedną zmianę szyku zdania i nie zauważyłem wcześniej tego czasownika "robić". A zatem, moja propozycja:
==> Dodatkowo zmiany wykonane na boku, mogą wpłynąć na jego późniejsze wykonanie za pomocą migracji, np. gdy wykonujemy CREATE TABLE bezpośrednio na bazie, a później dodajemy migrację schematu, która to procesuje, to w takim wypadku otrzymamy błąd informujący o tym, że taka tabela już istnieje.

Reply:
done